src/store/printer/actions.ts

```diff
@@ -152,15 +154,63 @@ export const actions: ActionTree<PrinterState, RootState> = {
         rootStateServerConfig?.server?.temperature_store_size ??
         Globals.CHART_HISTORY_RETENTION
       handleAddChartEntry(retention, getters.getChartableSensors)
+      dispatch('onDiagnosticsMetricsUpdate')
     }
   },
 
-  async onFastNotifyStatusUpdate ({ rootState, commit }, payload) {
+  async onFastNotifyStatusUpdate ({ rootState, commit, dispatch }, payload) {
     // Do NOT accept updates until our subscribe comes back.
     // This is because moonraker currently sends notification updates
     // prior to subscribing on browser refresh.
     if (payload && rootState.socket.acceptingNotifications) {
       commit('printer/setSocketNotify', payload, { root: true })
+      dispatch('onDiagnosticsMetricsUpdate')
     }
+  },
+
+  async onDiagnosticsMetricsUpdate ({ rootState, commit, rootGetters }) {
+    if (!rootState.config.uiSettings.general.enableDiagnostics) return
+    const layout = rootState.layout.layouts.diagnostics as DiagnosticsCardContainer
+    const metrics = Object.values(layout)
+      .flat()
+      .map(card => card.axes)
+      .flat()
+      .filter(axis => axis.enabled)
+      .map(axis => axis.metrics)
+      .flat()
+
+    const collectors = Array.from(new Set(metrics.map(metric => metric.collector)))
```

Comment:
Converting to a set and then back to an array deduplicates the entries (to prevent multiple eval calls for the same collector code which always yields the same result)

Reply:
Got it, all makes sense now!